flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlCreateTable.java

```diff
@@ -76,6 +76,8 @@ public class SqlCreateTable extends SqlCreate implements ExtendedSqlNode {
 
     private final boolean isTemporary;
 
+    private final SqlNode query;
```

Comment:
What do you think about introduce a new SqlNode `SqlCreateTableAs` which extends the `SqlCreate` and implement `ExtendedSqlNode`? In this node, we refer to the `SqlCreateTable` and `asQuery` as member, so we can separate the two different function node, it seems that more clear. 
Moreover, maybe we need hint in ctas syntax, such as `CREATE TABLE ... AS SELECT ... /*+ `OPTIONS`('key' = 'val') */`

Reply:
good catch!